From 85816410100ca0c694be63dff5a991bcd554a52f Mon Sep 17 00:00:00 2001
From: Maggie Cai <mxcai@chromium.org>
Date: Tue, 12 Nov 2019 22:28:12 +0000
Subject: [PATCH] [IntentHandling] Remove the DCHECK_EQ for unique_ptr in App
 Service.

This CL removes the DECHECK_EQ for the unique_ptr because this is
a C++20 feature.

BUG=957519, 980914, 853604

Change-Id: I87c3213788ac4447aad2f3d46f8af2876ebf0053
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/1910821
Reviewed-by: Nancy Wang <nancylingwang@chromium.org>
Reviewed-by: Raphael Kubo da Costa <raphael.kubo.da.costa@intel.com>
Commit-Queue: Maggie Cai <mxcai@chromium.org>
Cr-Commit-Position: refs/heads/master@{#714688}
---
 chrome/services/app_service/app_service_impl.cc | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/chrome/services/app_service/app_service_impl.cc b/chrome/services/app_service/app_service_impl.cc
index 3eee6dfde714f..221445f774c53 100644
--- a/chrome/services/app_service/app_service_impl.cc
+++ b/chrome/services/app_service/app_service_impl.cc
@@ -237,11 +237,10 @@ void AppServiceImpl::ConnectToPrefService(
 
 void AppServiceImpl::OnPrefServiceConnected(
     std::unique_ptr<PrefService> pref_service) {
-  if (!pref_service) {
+  if (!pref_service || pref_service_) {
     // TODO(crbug.com/853604): Handle if not successfully connected.
     return;
   }
-  DCHECK_EQ(nullptr, pref_service_);
   pref_service_ = std::move(pref_service);
   InitializePreferredApps();
 }
